subsys/bluetooth/host/keys.c

```diff
@@ -371,3 +412,26 @@ SETTINGS_STATIC_HANDLER_DEFINE(bt_keys, "bt/keys", NULL, keys_set, keys_commit,
 			       NULL);
 
 #endif /* CONFIG_BT_SETTINGS */
+
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+void bt_keys_notify_usage(u8_t id, const bt_addr_le_t *addr)
+{
+	if (!addr) {
```

Comment:
In what scenario would it *not* be a programming bug to pass `NULL` as `addr` to this function? If there's no such case then this test should be removed or alternatively changed into an assert (if you think there's risk of having such bugs). From the looks of it the address comes from `bt_conn_get_dst()` which is guaranteed to succeed if you pass it a valid `conn` object (it simply does `return &conn->le.dst;`) so I'd say you can just remove the NULL check here.